### DISCUSSION TITLE
Do we need Topics?

### DISCUSSION THREAD

**Discussion by rfc2822** [bitfireAT/webdav-push#24] [General]
Is the "collection URL" as notification message enough for clients to determine what has changed?

If a collection could be represented by two different URLs, the server could send another version in the push message than the client used to subscribe. Then the client can't reliably map the URL from the push message to a collection.

So collection URLs would have to be canonicalized. Also, collection URLs can only be used if end-to-end encryption is guaranteed.

Another method would be to have a "topic", some more-or-less random identifier, for every collection. The WebDAV server could send that topic as a WebDAV property:

```
PROPFIND /collection/
<propfind><prop><P:topic/>…

207 …
<topic>Roobubai9poi0ihe</topic>
```

In this case, `/collection/` would be assigned to `Roobubai9poi0ihe` by the server, and push messages would contain the topic instead of the (not necessarily unique) collection URL. If this topic would be sent over a non-E2EE transport, the push transport would still know which clients receive the same token, but it wouldn't know what it's about.

The token could also be used to replace not yet delivered push messages, if supported by the transport. See [`Topic` header in RFC 8030](https://www.rfc-editor.org/rfc/rfc8030#section-5.4).

**Comment by tcitworld** [comment]
Using topics makes sense to me indeed.
